File: algo/algo/완주하지못한선수.py
def solution(participant, completion):
    dic = {i:0 for i in set(participant)}
    for i in participant: dic[i] += 1
    for i in completion: dic[i] -= 1
    for i in dic.keys():
        if dic[i] != 0: return i

# # print(solution(["leo", "kiki", "eden"],["eden", "kiki"]))
# # print(solution(["marina", "josipa", "nikola", "vinko", "filipa"],["josipa", "filipa", "marina", "nikola"]))
print(solution(["mislav", "stanko", "mislav", "ana"],["stanko", "ana", "mislav"]))

File: algo/algo/Java/_2402/완주하지못한선수.java
import java.util.HashMap;

public class 완주하지못한선수 {
    public static void main(String[] args) {
        Solution s = new Solution();
        String[] participant = {"mislav", "stanko", "mislav", "ana"};
        String[] completion = {"stanko", "ana", "mislav"};
        System.out.println(s.solution(participant,completion));
    }
}

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String,Integer> hm = new HashMap<>();
        String result = "";
        for (String i: participant) {
            if (hm.containsKey(i)) hm.put(i,hm.get(i)+1);
            else hm.put(i,1);
        }
        for (String i: completion) hm.put(i,hm.get(i)-1);
        for (String i: hm.keySet()) {
            if (hm.get(i) != 0) {
                result = i;
                break;
            };
        }
        return result;
    }
}
